src/main/java/pl/connectis/spotifyapicli/authorization/AuthorizationCodeAuthorization.java

```diff
@@ -20,35 +20,29 @@
 
 @Slf4j
 @Component("authorization_code")
```

Comment:
zamiast nazywania beana za kazdym razem a potem wybor strategi poprzez wyszukanie odpowiedniej nazwy w app context użyj mechanizmow spring boota. `@ConditionalOnProperty(prefix = "auth", name = "authorizationMethod", value="authorizationCode")`
W drugiej klasie mozesz dac to samo albo:
`@ConditionalOnMissingBean(AuthorizationStrategy.class)`
To spowoduje ze jesli Spring nie stworzy beana AuthorizationCodeAuthorization (bo warunek wyzej nie bedzie spelniony) to wtedy stworzy ta druga